Documentation/bpf.rst

```diff
@@ -4,180 +4,184 @@
 BPF and XDP Reference Guide
 ***************************
 
-.. note:: This purely serves as a developer's guide and is explicitly not a
-          requirement for Cilium users to read.
-
-eBPF is a minimal, highly flexible and efficient "virtual machine"-like construct
-in the Linux kernel which is used in a number of subsystems, most prominently
+.. note:: This documentation section is targeted at developers and users who
+          want want to understand BPF and XDP in great technical depth. While
+          reading this reference guide may help broaden yur understanding of
+          Cilium, it is not a requirement to use Cilium. Please refer to the
+          :ref:`_gs_guide` and :ref:`_arch_guide` for a higher level
+          introduction.
+
+BPF is a highly flexible and efficient "virtual machine"-like construct in the
+Linux kernel allowing to execute bytecode at various hook points in a safe
+manner. It is used in a number of Linux kernel subsystems, most prominently
 networking, tracing and security (f.e. sandboxing).
 
-It replaced the traditional in-kernel "classic" BPF (cBPF) interpreter, which is
-perhaps mostly known from tcpdump filters that are passed as cBPF bytecode into
-the kernel. eBPF stands for "extended Berkeley Packet Filter", but has little to
-do with being tied to only packet filters these days. The instruction set is
-generic and flexible enough that there are many kernel subsystems which use eBPF
-apart from only networking. Nowadays, the Linux kernel runs eBPF only and loaded
-cBPF bytecode is transparently translated into an eBPF representation in the
-kernel before program execution.
-
-Note that, generally, the term "BPF" refers to eBPF implicitly today (and not
-cBPF anymore)!
-
-Cilium uses eBPF heavily in its data path, see :ref:`arch_guide` for further
-information. The goal of this chapter is to provide an eBPF reference guide in
-oder to gain basic understanding of eBPF and networking related eBPF program
-types such as tc (traffic control) and XDP (eXpress Data Path), and to aide
-developing Cilium's eBPF templates.
-
-The older cBPF architecture is not covered by this document, since Cilium does
-not make use of it.
-
-eBPF Architecture
-=================
-
-eBPF does not define itself by only providing its instruction set, but also by
+While BPF has existed since 1992, this document covers the version enhanced
+Berkley Paket Filter (eBPF) which has first appeared in Kernel 3.18 and
+obsoletes the traditional version which is being referred to as "classic" BPF
+(cBPF) these days. cBPF is known to many as being the packet filter language
+use by tcpdump.  Nowadays, the Linux kernel runs eBPF only and loaded cBPF
+bytecode is transparently translated into an eBPF representation in the kernel
+before program execution.
+
```

Comment:
I would still make an explicit sentence that when we talk about "BPF" we generally mean eBPF unless otherwise noted.